### ISSUE TITLE
Error when training a TSP

### ISSUE THREAD

**Issue by LuciaTajuelo** [wouterkool/attention-learn-to-route#38]
Hi:

I'm trying to train the model but i get an error when running `python run.py --graph_size 20 --baseline rollout --run_name 'tsp20_rollout'`

The trace log looks like:
```

$ python run.py --graph_size 20 --baseline rollout --run_name 'tsp20_rollout'
{'baseline': 'rollout',
 'batch_size': 512,
 'bl_alpha': 0.05,
 'bl_warmup_epochs': 1,      
 'checkpoint_encoder': False,
 'checkpoint_epochs': 1,     
 'data_distribution': None,  
 'embedding_dim': 128,       
 'epoch_size': 1280000,      
 'epoch_start': 0,
 'eval_batch_size': 1024,    
 'eval_only': False,
 'exp_beta': 0.8,
 'graph_size': 20,
 'hidden_dim': 128,
 'load_path': None,
 'log_dir': 'logs',
 'log_step': 50,
 'lr_critic': 0.0001,
 'lr_decay': 1.0,
 'lr_model': 0.0001,
 'max_grad_norm': 1.0,
 'model': 'attention',
 'n_encode_layers': 3,
 'n_epochs': 100,
 'no_cuda': False,
 'no_progress_bar': False,
 'no_tensorboard': False,
 'normalization': 'batch',
 'output_dir': 'outputs',
 'problem': 'tsp',
 'resume': None,
 'run_name': 'tsp20_rollout_20210507T142303',
 'save_dir': 'outputs\\tsp_20\\tsp20_rollout_20210507T142303',
 'seed': 1234,
 'shrink_size': None,
 'tanh_clipping': 10.0,
 'use_cuda': False,
 'val_dataset': None,
 'val_size': 10000}
Evaluating baseline model on evaluation dataset
check2
  0%|                                                                                                                                                                                                                               | 0/10 [00:00<?, ?it/sT 
raceback (most recent call last):
  File "run.py", line 172, in <module>
    run(get_options())
  File "run.py", line 104, in run
    baseline = RolloutBaseline(model, problem, opts)
  File "C:\Users\lucia\Desktop\Trucksters-repo\atention_model\reinforce_baselines.py", line 151, in __init__
    self._update_model(model, epoch)
  File "C:\Users\lucia\Desktop\Trucksters-repo\atention_model\reinforce_baselines.py", line 171, in _update_model
    self.bl_vals = rollout(self.model, self.dataset, self.opts).cpu().numpy()
  File "C:\Users\lucia\Desktop\Trucksters-repo\atention_model\train.py", line 43, in rollout
    in tqdm(DataLoader(dataset, batch_size=opts.eval_batch_size), disable=opts.no_progress_bar)
  File "C:\Users\lucia\Desktop\Trucksters-repo\atention_model\train.py", line 42, in <listcomp>
    for bat
  File "C:\Users\lucia\Desktop\Trucksters-repo\atention_model\train.py", line 37, in eval_model_bat
    cost, _ = model(move_to(bat, opts.device))
  File "C:\Users\lucia\.virtualenvs\atention_model-OuevcIBs\lib\site-packages\torch\nn\modules\module.py", line 889, in _call_impl
    result = self.forward(*input, **kwargs)
  File "C:\Users\lucia\Desktop\Trucksters-repo\atention_model\nets\attention_model.py", line 137, in forward
    _log_p, pi = self._inner(input, embeddings)
  File "C:\Users\lucia\Desktop\Trucksters-repo\atention_model\nets\attention_model.py", line 234, in _inner
    batch_size = state.ids.size(0)
  File "C:\Users\lucia\Desktop\Trucksters-repo\atention_model\problems\tsp\state_tsp.py", line 31, in __getitem__
    assert torch.is_tensor(key) or isinstance(key, slice)  # If tensor, idx all tensors by this tensor:
AssertionError
  0%|                                                                                                                                                                                                                               | 0/10 [00:00<?, ?it/s]
```

How can i get rid of it?

Thanks in advance

**Comment by pataq2** [comment]
Hi:

I have enjoyed a lot reading your paper, thank you very much! I had the same issue. I look forward a response. 

Thanks in advance!

**Comment by wouterkool** [comment]
Hi! Are you using Python >= 3.8?

**Comment by LuciaTajuelo** [comment]
Solved upgrading to Python 3.8.
Thanks a lot!
